third_party/ros/rostest/BUILD.bazel

```diff
@@ -12,45 +13,69 @@ py_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        "@com_github_mvukov_rules_ros//third_party/ros:roslaunch",
         "@ros_comm//:rospy",
         "@ros_ros//:rosunit",
+        "@rules_ros//third_party/ros:roslaunch",
     ],
 )
 
 py_binary(
-    name = "advertisetest",
+    name = "advertisetest_impl",
     srcs = ["advertisetest.py"],
-    visibility = ["//visibility:public"],
+    main = "advertisetest.py",
     deps = [
         ":rostest",
-        "@com_github_mvukov_rules_ros//third_party/ros/rosservice",
+        "@rules_ros//third_party/ros/rosservice",
     ],
 )
 
+symlink(
```

Comment:
I don't like to approach in particular, but until py_binary starts to have a single file in DefaultInfo, I am afraid we'll need this.

In particular: https://github.com/bazelbuild/rules_python/blob/main/python/private/common/py_executable.bzl#L767: DefaultInfo.files should be just the executable, the rest should go to runfiles. For backwards compatibility reasons, this functionality needs to go under a new incompatible flag.